hat/assets/js/apps/Iaso/domains/instances/hooks/useGetProfilesDropdown.tsx

```diff
@@ -0,0 +1,34 @@
+import { UseQueryResult } from 'react-query';
+import { useSnackQuery } from '../../../libs/apiHooks';
+import { DropdownOptions } from '../../../types/utils';
+import MESSAGES from '../messages';
+import getDisplayName, { Profile } from '../../../utils/usersUtils';
+import { getRequest } from '../../../libs/Api';
+import { makeUrlWithParams } from '../../../libs/utils';
+
+export const useGetProfilesDropdown = (
+    ids: string,
+): UseQueryResult<DropdownOptions<number>, Error> => {
+    return useSnackQuery(
+        ['profiles', ids],
+        () => {
+            if (ids === undefined || ids.length < 1) {
+                return [];
+            }
+            return getRequest(makeUrlWithParams('/api/profiles/', { ids }));
+        },
+        MESSAGES.error,
+        {
+            select: data => {
+                return (
+                    data?.profiles?.map((profile: Profile) => {
+                        return {
+                            value: profile.user_id,
+                            label: getDisplayName(profile),
+                        };
+                    }) ?? []
+                );
+            },
+        },
+    );
```

Comment:
This is much more readable if you pass an object instead of an arrays of props, `useSnackQuery` support both